### DISCUSSION TITLE
Objeck's for loop

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#144] [General]
I always wondered why there need to be a semicolon before `)`

```
location := "East Bay";
for(i := 0; i < location->Size(); i += 1;) {
 location->Get(i)->PrintLine();
}; 
```

On other C based languages, it will be:

`for(i := 0; i < location->Size(); i += 1)`

Objeck's for loop is weird.

Btw, on Operators table, I found it support `++`. But I have never seen a for loop that uses `i++` or `++i`.

**Comment by objeck** [comment]
Here's the [rational](https://github.com/objeck/objeck-lang/blob/master/core/compiler/parser.cpp#L4856), a for loop update statement is itself a statement; statements end in semi-colons. 

That said, the code is [stubbed out](https://github.com/objeck/objeck-lang/blob/master/core/compiler/parser.cpp#L1433) to make it optional however I have not implemented this option.

**Reply by objeck** [reply]
Now optional without the use of the alternative syntax. 8d8c82ba8

**Reply by ghost** [reply]
Please update the language server, it still gives errors if don't have the semicolon at the end of block statements @objeck

**Reply by objeck** [reply]
Are you using v2023.2.2 of the tool chain and language servers?

**Reply by ghost** [reply]
Sorry, I forgot to update to latest version :smile:

**Comment by objeck** [comment]
No problem, I just remembered that I need to add that feature to the release notes. It is in the upcoming release notes.
